core/rpc/src/system/mod.rs

```diff
@@ -115,4 +119,15 @@ impl<B: traits::Block> SystemApi for System<B> {
 			Ok(health)
 		}
 	}
+
+	fn system_peers(&self) -> Result<Vec<PeerInfo>> {
+		Ok(self.sync.peers().into_iter().map(|(idx, peer_id, p)| PeerInfo {
+			index: idx as usize,
+			peer_id: peer_id.map_or("".into(), |p| p.to_base58()),
```

Comment:
`map_or_default`

Reply:
```suggestion
			peer_id: peer_id.map_or_default(|p| p.to_base58()),
```

Reply:
there seems no `map_or_default` method?

Reply:
That's better - no literal and no constructor called in case where it's `Some`

Reply:
```suggestion
			peer_id: peer_id.map_or_else(Default::default, |p| p.to_base58()),
```